src/Umbraco.Core/Logging/DisposableTimer.cs

```diff
@@ -85,19 +118,55 @@ protected override void DisposeResources()
             {
                 if (_failed)
                 {
-                    _logger.LogError(_failException, "{FailMessage} ({Duration}ms) [Timing {TimingId}]", _failMessage, Stopwatch.ElapsedMilliseconds, _timingId);
+                    if (_failMessageArgs == null)
+                    {
+                        _logger.LogError(_failException, "{FailMessage} ({Duration}ms) [Timing {TimingId}]", _failMessage, Stopwatch.ElapsedMilliseconds, _timingId);
+                    }
+                    else
+                    {
+                        var args = new object[_failMessageArgs.Length + 2];
+                        _failMessageArgs.CopyTo(args, 0);
+                        args[_failMessageArgs.Length - 1] = Stopwatch.ElapsedMilliseconds;
+                        args[_failMessageArgs.Length] = _timingId;
```

Comment:
Seems like we're doing more or less the same thing three times, maybe wrap this in a method? Maybe just pass in the  `object[]`, and it can be used for both failed and succeeded.